distribution/src/bin/elasticsearch

```diff
@@ -54,6 +54,14 @@ fi
 #   - fourth, ergonomic JVM options are applied
 ES_JAVA_OPTS=`export ES_TMPDIR; "$JAVA" "$XSHARE" -cp "$ES_CLASSPATH" org.elasticsearch.tools.launchers.JvmOptionsParser "$ES_PATH_CONF"`
 
+# If the quote-aware filesystem plugin is installed, then we need to pass extra
+# flags to Java in order to use the plugin.
+if [[ -d "$ES_HOME/plugins/quota-aware-fs" ]]; then
```

Comment:
The idea of bootstrap plugins is to avoid this type of specialized logic (https://github.com/elastic/elasticsearch/issues/29306). Instead, the plugin should define which system properties need to be added, and bin/elasticsearch should introspect those upon finding all bootstrap plugins.

Reply:
I've added infrastructure to do this, see what you think.

What I haven't done, however, is update `bin/elasticsearch.bat` (yet). My batch knowledge it pretty woeful, so if we want to support this mechanism on Windows, I'll need to pair with someone who knows what they're doing 😅